test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HeaderUserIdProvider.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.SignalR.Client.FunctionalTests
+{
+    internal class HeaderUserIdProvider : IUserIdProvider
+    {
+        public static readonly string HeaderName = "Super-Secure-UserName";
+
+        public string GetUserId(HubConnectionContext connection)
+        {
+            // Super-secure user id provider :)
```

Comment:
Don't let Barry see this 😄

Reply:
I'll rename the header and comments just in case someone doesn't understand my sarcasm ;)